samples/cpp/gstreamer_pipeline.cpp

```diff
@@ -36,12 +35,22 @@ class GStreamerPipeline
         fast_measure = cmd_parser->has("fast");               // fast measure fps
         fix_fps      = cmd_parser->get<int>("fps");           // fixed frame per second
         pipeline     = cmd_parser->get<string>("pipeline"),   // gstreamer pipeline type
-        container    = cmd_parser->get<string>("container"),  // container type
         mode         = cmd_parser->get<string>("mode"),       // coding mode
         codec        = cmd_parser->get<string>("codec"),      // codec type
         file_name    = cmd_parser->get<string>("file"),       // path to videofile
         resolution   = cmd_parser->get<string>("resolution"); // video resolution
 
+        std::size_t found = file_name.rfind(".");
+        if (found != std::string::npos)
+        {
+            container = file_name.substr(found + 1);  // container type
+        }
+        else
+        {
+            std::cout << "Can not parse container extension" << '\n';
+            exit_code = -1;
```

Comment:
`exit_code = -1` is useless here, because it is overrided on line 59.
The same problem is with lines 32, 57.

"exit_code" should be removed (anywhere) and replaced to `throw std::exception("message")` or early return.